### PR TITLE
[lldb] Use Python script to generate SBLanguages.h

### PR DESCRIPTION
**Pull request by JDevlieghere**
Use a Python script to generate SBLanguages.h instead of piggybacking on LLDB TableGen. This addresses Nico Weber's post-commit feedback.

### CODE REVIEW

File: lldb/scripts/generate-sbapi-dwarf-enum.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        # Emit the header.
+        f.write(HEADER)
+
+        # Emit the enum values.
+        for line in lines:
+            match = REGEX.search(line)
+            if not match:
+                continue
+            f.write(f"  /// {match.group(4)}.\n")
+            f.write(f"  eLanguageName{match.group(3)} = {match.group(2)},\n")
```
Comment (kastiglione) [inline lldb/scripts/generate-sbapi-dwarf-enum.py:49]: these would benefit from using named captures, ex:

```suggestion
            f.write(f"  /// {match.group("whatever_group_4_is")}.\n")
            f.write(f"  eLanguageName{match.group("name")} = {match.group("value")},\n")
```

File: lldb/scripts/generate-sbapi-dwarf-enum.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+enum SBSourceLanguageName : uint16_t {
+"""
+
+FOOTER = """\
+};
+
+#endif
+"""
+
+REGEX = re.compile(r'(^ *HANDLE_DW_LNAME *\( *([^,]+), ([^,]+), )"(.*)",.*\).*')
```
Comment (kastiglione) [inline lldb/scripts/generate-sbapi-dwarf-enum.py:29]:
```suggestion
REGEX = re.compile(r'^ *HANDLE_DW_LNAME *\( *(?P<value>[^,]+), (?P<ident>[^,]+), "(?P<name>.*)",.*\)')
```

File: lldb/scripts/generate-sbapi-dwarf-enum.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+"""
+
+FOOTER = """\
+};
+
+#endif
+"""
+
+REGEX = re.compile(
+    r'^ *HANDLE_DW_LNAME *\( *(?P<value>[^,]+), (?P<comment>[^,]+), "(?P<name>.*)",.*\)'
```
Comment (kastiglione) [inline lldb/scripts/generate-sbapi-dwarf-enum.py:30]: looks like you have comment and name swapped

**Review by adrian-prantl — APPROVED** [review]
Thanks!

**Review by adrian-prantl — CHANGES_REQUESTED** [review]

**Review by adrian-prantl — APPROVED** [review]

File: lldb/scripts/generate-sbapi-dwarf-enum.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+"""
+
+FOOTER = """\
+};
+
+#endif
+"""
+
+REGEX = re.compile(
+    r'^ *HANDLE_DW_LNAME *\( *(?P<value>[^,]+), (?P<comment>[^"]+), "(?P<name>.*)",.*\)'
```
Comment (kastiglione) [inline lldb/scripts/generate-sbapi-dwarf-enum.py:30]: to combine my feedback with Adrian's:

```python
r'^ *HANDLE_DW_LNAME *\( *(?P<value>[^,]+), (?P<name>[^,]+), "(?P<comment>[^"]*)",.*\)'
```

File: lldb/source/API/CMakeLists.txt [hunk 5]
```diff
@@ -20,9 +20,19 @@ if(LLDB_ENABLE_LUA)
   set(lldb_lua_wrapper ${lua_bindings_dir}/LLDBWrapLua.cpp)
 endif()
 
-lldb_tablegen(../../include/lldb/API/SBLanguages.h -gen-lldb-sbapi-dwarf-enum
-  SOURCE ${LLVM_MAIN_INCLUDE_DIR}/llvm/BinaryFormat/Dwarf.def
-  TARGET lldb-sbapi-dwarf-enums)
+# Target to generate SBLanguages.h from Dwarf.def.
+set(sb_languages_file
+  ${CMAKE_CURRENT_BINARY_DIR}/../../include/lldb/API/SBLanguages.h)
+add_custom_target(
```
Comment (labath) [inline lldb/source/API/CMakeLists.txt:26]: I believe this should be using `add_custom_command`. Per the [cmake docs](https://cmake.org/cmake/help/latest/command/add_custom_target.html), `add_custom_target` "... has no output file and is always considered out of date even if the commands try to create a file with the name of the target". This causes an incremental clean build to always run this command and also rebuild anything that depends on it.

Comment (JDevlieghere) [reply lldb/source/API/CMakeLists.txt:26]: https://github.com/llvm/llvm-project/pull/91254

**Pull request merged**